src/test/java/com/artipie/api/UsersRestTest.java

```diff
@@ -82,6 +87,82 @@ vertx, ctx, new TestRequest("/api/v1/users/Jane"),
         );
     }
 
+    @Test
+    void replacesUser(final Vertx vertx, final VertxTestContext ctx) throws Exception {
+        this.save(
+            new Key.From(ManageUsersTest.KEY),
+            new CredsConfigYaml().withUsers("Mark")
+                .withFullInfo(
+                    "John", Users.PasswordFormat.PLAIN, "231", "john@example.com",
+                    Set.of("readers", "tags")
+                ).toString().getBytes(StandardCharsets.UTF_8)
+        );
+        this.requestAndAssert(
+            vertx, ctx, new TestRequest(
+                HttpMethod.PUT, "/api/v1/users/Mark",
+                new JsonObject().put("type", "plain").put("pass", "qwerty")
+                    .put("email", "mark@example.com")
+            ),
+            response -> {
+                MatcherAssert.assertThat(
+                    response.statusCode(),
+                    new IsEqual<>(HttpStatus.CREATED_201)
+                );
+                MatcherAssert.assertThat(
+                    new String(this.storage().value(ManageUsersTest.KEY), StandardCharsets.UTF_8),
+                    new StringContains(
+                        String.join(
+                            System.lineSeparator(),
+                            "  Mark:",
+                            "    type: plain",
+                            "    pass: qwerty",
+                            "    email: mark@example.com"
+                        )
+                    )
+                );
+            }
+        );
+    }
+
+    @Test
+    void addsUser(final Vertx vertx, final VertxTestContext ctx) throws Exception {
+        this.save(
+            new Key.From(ManageUsersTest.KEY),
+            new CredsConfigYaml().withUsers("Mark").toString().getBytes(StandardCharsets.UTF_8)
```

Comment:
Why do we need "Mark" here?
If I remove it, nothing changes.
If we want to check cases when existed users can affect some operations, we should explicitly create tests for these cases.